### ISSUE TITLE
load genom by selecting file ? 

### ISSUE THREAD

**Issue by dridk** [BIG-Tools/BigBrowser#55]
as @natir suggest, it's a bad idea, to have one folder for genomes. 
Create from the Menu :  Genom > open genom,   Annotation > open annotation

**Comment by dridk** [comment]
At the end, I m not sure it's a good idea. 
We are going to preprocess file and create associated index file. 
Also, from the user side, it's easier to have one place to store everything. 
For instance, steam application, download and manage everything from the same place. 
Let's discuss about it

**Comment by dridk** [comment]
In the other hand, I do not want to import my annotation file ... !

**Comment by dridk** [comment]
Let's do like IGV ! 
Genome file are imported into a common folder. 
But annotation a read directly from their location.

**Comment by natir** [comment]
I don't understand why bigbroswer required a special directory for file.
File generate by bigbroswer is save is same directory of original file.

I didn't see the problem and create a bigbroswer directory in home directory of the user is disgusting.

**Comment by dridk** [comment]
I partially agree.
When I created the folder path, I was thinking of a download manager  which
download genom and annotation from server.
When application start, It loads available genom name and display them into
the genom combobox. To do that,  I have to scan a defined folder.

I think you are right for annotation local file,  which can reside
anywhere.
But what about downloaded file from bigbrother? Where must they go?

To summarise,  they are 2 kind of data.
- genome package from local or network
- annotation from local or network

When you click on AddData, a dialog box should appear which allows user to
select data from local or from server

**Comment by natir** [comment]
If the question is where store the download file ? The answer is in same for all application in the download directory (but user can set this parameter).

We can use this to find it http://doc.qt.io/qt-5/qstandardpaths.html

**Comment by dridk** [comment]
What about genomes files? And how to make application understand that some
annotation file works only with a genom file .

2016-04-02 9:28 GMT+02:00 Pierre Marijon notifications@github.com:

> If the question is where store the download file ? The answer is in same
> for all application in the download directory (but user can set this
> parameter).
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/BIG-Tools/BigBrowser/issues/55#issuecomment-204666316

## 

## 

Sacha schutz
Interne en Biologie moléculaire
Bioinformaticien
+33651571606
dridk.me
Fork me on github https://github.com/

**Comment by natir** [comment]
It's the user responsibility to be sure the annotation file match with sequence file.

**Comment by dridk** [comment]
And about download folder,
what about having annotationDownloadFolder , with subfolder ordered by
categories.

annotation / hg19 / gcPercent.bed

2016-04-02 11:20 GMT+02:00 Pierre Marijon notifications@github.com:

> It's the user responsibility to be sure the annotation file match with
> sequence file.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/BIG-Tools/BigBrowser/issues/55#issuecomment-204679236

## 

## 

Sacha schutz
Interne en Biologie moléculaire
Bioinformaticien
+33651571606
dridk.me
Fork me on github https://github.com/

**Comment by natir** [comment]
For your example, User add track and after in the track setting select file we want

I didn't understand why you want a strict subfolder organization let the user to be free. I hate application we force me to use special file organization. Application work for me, I didn't work for application.

**Comment by dridk** [comment]
I understand your point.
But In most of the case, user will download items from download manager. I
m talking about this point.

2016-04-02 11:37 GMT+02:00 Pierre Marijon notifications@github.com:

> For your example, User add track and after in the track setting select
> file we want
> 
> I didn't understand why you want a strict subfolder organization let the
> user to be free. I hate application we force me to use special file
> organization. Application work for me, I didn't work for application.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/BIG-Tools/BigBrowser/issues/55#issuecomment-204682646

## 

## 

Sacha schutz
Interne en Biologie moléculaire
Bioinformaticien
+33651571606
dridk.me
Fork me on github https://github.com/

**Comment by natir** [comment]
Yes and file is download in system download dir, and if user want move/remove/modified file BB  is ok. 

In your approach BB is good for one use case : 
- User download file with BB
- User see file with BB

In my approach BB is good for all more use case :
- User download/build/create file with any software
- User see any file with BB

**Comment by dridk** [comment]
So let's do annotation as you said .
But genom should be application file.

2016-04-02 12:05 GMT+02:00 Pierre Marijon notifications@github.com:

> Yes and file is download in system download dir, and if user want
> move/remove/modified file BB is ok.
> 
> In your approach BB is good for one use case :
> - User download file with BB
> - User see file with BB
> 
> In my approach BB is good for all more use case :
> - User download/build/create file with any software
> - Use see any file with BB
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/BIG-Tools/BigBrowser/issues/55#issuecomment-204685737

## 

## 

Sacha schutz
Interne en Biologie moléculaire
Bioinformaticien
+33651571606
dridk.me
Fork me on github https://github.com/
